### PR TITLE
[kotlin] J2K: Convert RemoveRedundantModalityModifierProcessing post-processing to a JK Tree printing phase

### PR DESCRIPTION
**Pull request by jocelynluizzi13**
Move RemoveRedundantModalityModifierProcessing to JKTree printing.

Additionally add interface checks in the ConvertGettersAndSettersToPropertyProcessing postprocessing step since this is where some of the properties get created and ABSTRACT is added. This will be migrated during group 2

@abelkov @darthorimar @ermattt

### CODE REVIEW

**Comment by abelkov** [comment]
Please use Git rebase to not introduce redundant merge commits. Roughly speaking, https://patch-diff.githubusercontent.com/raw/JetBrains/intellij-community/pull/2648.patch should be very small and show only the changes actually made by you. This will greatly ease the process of merging PRs. In the end, this PR should consist of exactly one commit with your changes.

File: plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt [hunk 1]
```diff
@@ -965,4 +964,4 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
             printer.print(label.render())
         }
     }
-}
+}
```
Comment (abelkov) [inline plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt:967]: Not needed change

File: plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt [hunk 2]
```diff
@@ -64,14 +63,14 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
...
+            val isOpenAndAbstractByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
                     (modifiersListOwner is JKMethod && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
+                    || (modifiersListOwner is JKField && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
             val hasOverrideModifier = modifiersListOwner
                 .safeAs<JKOtherModifiersOwner>()
                 ?.hasOtherModifier(OVERRIDE) == true
-
             fun Modifier.isRedundant(): Boolean = when {
-                this == OPEN && isOpenByDefault -> true
+                (this == OPEN || this == ABSTRACT) && isOpenAndAbstractByDefault -> true
```
Comment (abelkov) [inline plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt:73]: I'm a little confused. We have both "or" and "and" here. Which one is it? Maybe it's better to split this into two branches?

File: plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt [hunk 3]
```diff
@@ -64,14 +63,14 @@ internal class JKCodeBuilder(context: NewJ2kConverterContext) {
         }
 
         private fun renderModifiersList(modifiersListOwner: JKModifiersListOwner) {
-            val isOpenByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
+            val isOpenAndAbstractByDefault = (modifiersListOwner is JKClass && modifiersListOwner.classKind == INTERFACE) ||
                     (modifiersListOwner is JKMethod && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
+                    || (modifiersListOwner is JKField && modifiersListOwner.parentOfType<JKClass>()?.classKind == INTERFACE)
```
Comment (abelkov) [inline plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt:68]: The initializer is getting complicated, probably it's worth extracting common subexpressions and/or simplifying in other ways.

**Review by abelkov — CHANGES_REQUESTED** [review]

File: plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt (hunk 2 above)
Comment (jocelynluizzi13) [reply plugins/kotlin/j2k/k1.new/src/org/jetbrains/kotlin/nj2k/printing/JKCodeBuilder.kt]: the interface would be both open and abstract by default, so if it has either modifier we should remove it- i can separate it into two branches if that makes it clearer

**Pull request closed without merging**